subsys/bluetooth/mesh/scheduler_srv.c

```diff
@@ -718,9 +732,33 @@ static void scheduler_srv_reset(struct bt_mesh_model *model)
 	net_buf_simple_reset(srv->pub.msg);
 }
 
+#ifdef CONFIG_BT_SETTINGS
+static int scheduler_srv_settings_set(struct bt_mesh_model *model,
+				      const char *name,
+				      size_t len_rd, settings_read_cb read_cb,
+				      void *cb_data)
+{
+	struct bt_mesh_scheduler_srv *srv = model->user_data;
+	struct bt_mesh_schedule_entry data;
+	ssize_t len = read_cb(cb_data, &data, sizeof(data));
+	uint8_t idx = strtol(name, NULL, 0);
+
+	if (len < sizeof(data) || idx >= BT_MESH_SCHEDULER_ACTION_ENTRY_COUNT) {
+		return -EINVAL;
+	}
+
+	srv->sch_reg[idx] = data;
+
+	return 0;
+}
+#endif
+
 const struct bt_mesh_model_cb _bt_mesh_scheduler_srv_cb = {
 	.init = scheduler_srv_init,
 	.reset = scheduler_srv_reset,
```

Comment:
Remember to erase model's data on `reset` callback.

Reply:
This is updated now.